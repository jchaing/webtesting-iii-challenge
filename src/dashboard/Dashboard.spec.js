// Test away

import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Dashboard from './Dashboard';

test('<Dashboard /> snapshot', () => {
  const wrapper = rtl.render(<Dashboard />);
  expect(wrapper.asFragment()).toMatchSnapshot();
});

test('Defaults to Unlocked and Open', () => {
  const { getByText } = rtl.render(<Dashboard />);

  expect(getByText(/unlocked/i).textContent).toMatch(/unlocked/i);
  expect(getByText(/open/i).textContent).toMatch(/open/i);
});

test('Shows the controls and display', () => {
  const { getByText } = rtl.render(<Dashboard />);

  expect(getByText(/unlocked/i).textContent).toMatch(/unlocked/i);
  expect(getByText(/open/i).textContent).toMatch(/open/i);
  expect(getByText(/lock gate/i).textContent).toMatch(/lock gate/i);
  expect(getByText(/close gate/i).textContent).toMatch(/close gate/i);
});

test('Cannot be Closed or Opened if it is Locked', () => {
  const wrapper = rtl.render(<Dashboard />);
  const locked = wrapper.queryByText(/unlock gate/i);

  locked && expect(wrapper.getByText(/locked/i).className).toMatch(/red-led/i);
  locked && expect(wrapper.getByText(/closed/i).className).toMatch(/red-led/i);
});
