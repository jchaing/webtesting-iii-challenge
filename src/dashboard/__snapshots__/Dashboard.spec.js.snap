// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="unlocked-locked"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="open-closed"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="key"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="entry"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
